learning_journal/tests.py

```
@@ -1,65 +1,249 @@
+    assert result["entries"][0].title == "one"
+
+
+def test_homepage_returns_empty_when_empty(dummy_request):
+    """Test that the list view returns no objects in the expenses iterable."""
+    from learning_journal.views.default import my_view
+    result = my_view(dummy_request)
+    assert len(result["entries"]) == 0
+
+def test_homepage_returns_existing_entries(dummy_request, add_models):
+    """Test that the list view serves up journal entries."""
+    from learning_journal.views.default import my_view
+    result = my_view(dummy_request)
+    assert len(result["entries"]) == len(TEST_ENTRIES)
+
+
+def test_detail_view(dummy_request, db_session):
+    """Test the detail view."""
+    from .views.default import detail
+    dummy_request.method = "POST"
+    dummy_request.POST["title"] = "This is a Title"
+    dummy_request.POST["body"] = "And this is the body. TEST!"
+    dummy_request.matchdict['id'] = '1'
+    result = detail(dummy_request)
+    entry = dummy_request.dbsession.query(Entry).get(1)
+    assert result['entry'] == entry
+
+
+def test_detail_page_returns_empty_when_empty(dummy_request):
+    """Test that the detail view returns no objects when database empty."""
+    from learning_journal.views.default import detail
+    req = dummy_request
+    req.matchdict = {"id": "1"}
+    result = detail(req)
+    assert result["entry"] is None
+
+
+def test_edit_page_updates_db(dummy_request, add_models):
+    """Test that the edit view edits entries in the database."""
+    from learning_journal.views.default import edit
+    req = dummy_request
+    req.matchdict = {"id": "3"}
+    req.method = "POST"
+    req.POST["title"] = "new thing"
+    req.POST["creation_date"] = datetime.datetime(2017, 2, 24, 0, 0)
+    req.POST["body"] = "and more new things"
+    try:
+        edit(req)
+    except:
+        new_title = dummy_request.dbsession.query(Entry).get(3).title
+        assert new_title == "new thing"
```

I'm severely confused about this. Why is there a bare except here?

---

I'm not sure what I was going for there; got rid of the try-except, dedented 175-176, and happy to see it working